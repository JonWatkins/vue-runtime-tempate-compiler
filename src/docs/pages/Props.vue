<template>
  <div class="page">
    <h1 id="top" class="mb-3">
      <router-link to="/guide/props#top" class="header-anchor">
        <span class="header-anchor--hash">#</span>
        Props
      </router-link>
    </h1>

    <p>
      Just like the <code>computed</code> the <code>props</code> properties are
      available in the <code>runtime-template-compiler</code>. The same as the
      other parent property you use them as if you are writing bindings in the
      parent's compiled template.
    </p>

    <pre class="language-html mb-4"><code>{{ example | trim }}</code></pre>

    <PageNav>
      <template v-slot:left>
        <router-link to="/guide/computed">
          <i class="fas fa-arrow-left"></i>
          Computed
        </router-link>
      </template>
      <template v-slot:right>
        <router-link to="/guide/provide">
          Provide
          <i class="fas fa-arrow-right"></i>
        </router-link>
      </template>
    </PageNav>
  </div>
</template>

<script>
import PageNav from '../components/PageNav.vue'

const example = `
<template>
  <runtime-template-compiler :template="template" \/>
<\/template>

<script>
import { RuntimeTemplateCompiler } from 'vue-runtime-template-compiler'

export default {
  components: {
    RuntimeTemplateCompiler
  },
  props: {
    firstName: {
      type: String,
      default: 'Bob'
    },
    lastName: {
      type: String,
      default: 'Example'
    }
  }
  data: () => ({
    template: \`<div>{{ greeting }}!<\/div>\`,
  }),
  computed: {
    fullName() {
      return \`\${this.firstName} \${this.lastName}\`
    }
  }
}
<\/script>
`

export default {
  name: 'Props',

  components: {
    PageNav
  },

  data() {
    return {
      example
    }
  }
}
</script>
