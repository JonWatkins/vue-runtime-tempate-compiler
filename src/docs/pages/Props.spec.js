import Props from './Props.vue'
const { shallowMountLocal } = global

let wrapper

describe('Props', () => {
  beforeEach(() => {
    wrapper = shallowMountLocal(Props)
  })

  it('is called', () => {
    expect(wrapper.name()).toBe('Props')
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
