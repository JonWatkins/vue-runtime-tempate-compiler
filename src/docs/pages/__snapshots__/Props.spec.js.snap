// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Props render correctly 1`] = `
<div class="page">
  <h1 id="top" class="mb-3">
    <router-link-stub to="/guide/props#top" tag="a" event="click" class="header-anchor"><span class="header-anchor--hash">#</span>
      Props
    </router-link-stub>
  </h1>
  <p>
    Just like the <code>computed</code> the <code>props</code> properties are
    available in the <code>runtime-template-compiler</code>. The same as the
    other parent property you use them as if you are writing bindings in the
    parent's compiled template.
  </p> <pre class="language-html mb-4"><code>&lt;template&gt;
  &lt;runtime-template-compiler :template="template" /&gt;
&lt;/template&gt;

&lt;script&gt;
import { RuntimeTemplateCompiler } from 'vue-runtime-template-compiler'

export default {
  components: {
    RuntimeTemplateCompiler
  },
  props: {
    firstName: {
      type: String,
      default: 'Bob'
    },
    lastName: {
      type: String,
      default: 'Example'
    }
  }
  data: () =&gt; ({
    template: \`&lt;div&gt;{{ greeting }}!&lt;/div&gt;\`,
  }),
  computed: {
    fullName() {
      return \`\${this.firstName} \${this.lastName}\`
    }
  }
}
&lt;/script&gt;</code></pre>
  <pagenav-stub></pagenav-stub>
</div>
`;
